.evergreen/config.yml

```diff
@@ -1071,6 +1071,25 @@ functions:
               --ca_file "${PROJECT_DIRECTORY}/testdata/kmip-certs/ca-ec.pem" \
               --cert_file "${PROJECT_DIRECTORY}/testdata/kmip-certs/server-ec.pem"
           fi
+    # Wait up to 10 seconds for the KMIP server to start.
+    - command: shell.exec
+      params:
+        script: |
+          cd ${DRIVERS_TOOLS}/.evergreen/csfle
+          if [ "Windows_NT" = "$OS" ]; then
+            export PYTHON=kmstlsvenv/Scripts/python.exe
+          else
+            export PYTHON=./kmstlsvenv/bin/python3
+          fi
+          for i in $(seq 1 1 10); do
+             sleep 1
+             if $PYTHON -u kms_kmip_client.py; then
+                echo 'KMS KMIP server started!'
```

Comment:
This may already be accomplished in some cases by the `await_server` check in [run-tests.sh](https://github.com/mongodb/mongo-go-driver/blob/41ec8764964a30c5b78a3136b4024603be5c11fe/.evergreen/run-tests.sh#L85). However, not every task uses `run-tests.sh`, so maybe this is a better place to add it.

If this and `await_server` try to accomplish the same thing, we should try to unify the logic and keep them in the same place.

Reply:
Thanks for the heads up, I consolidated the logic.